Remora.Commands/Builders/CommandParameterBuilder.cs

```diff
@@ -0,0 +1,453 @@
+    public CommandParameterBuilder IsOption(OneOf<char, string, (char Short, string Long)> shortOrLongName = default)
+    {
+        var canBeOption = !_attributes.Any(r => r is SwitchAttribute);
+
+        if (!canBeOption)
+        {
+            throw new InvalidOperationException("A parameter marked as an option cannot be a switch.");
+        }
+
+        var attribute = shortOrLongName.Match
+        (
+            shortName => new OptionAttribute(shortName),
+            longName => new OptionAttribute(longName),
+            shortAndLongName => new OptionAttribute(shortAndLongName.Short, shortAndLongName.Long)
+        );
+
+        _attributes.Add(attribute);
+
+        return this;
+    }
+
+    /// <summary>
+    /// Adds an attribute to the parameter. Conditions must be added via <see cref="AddCondition"/>.
+    /// </summary>
+    /// <param name="attribute">The attriubte to add.</param>
+    /// <returns>The builder to chain calls with.</returns>
+    public CommandParameterBuilder AddAttribute(Attribute attribute)
+    {
+        if (attribute is ConditionAttribute)
+        {
+            throw new InvalidOperationException("Conditions must be added via AddCondition.");
+        }
+
+        _attributes.Add(attribute);
+        return this;
+    }
+
+    /// <summary>
+    /// Adds a condition to the parameter.
+    /// </summary>
+    /// <param name="condition">The condition to add.</param>
+    /// <returns>The builder to chain with.</returns>
+    public CommandParameterBuilder AddCondition(ConditionAttribute condition)
+    {
+        _conditions.Add(condition);
+        return this;
+    }
+
+    /// <summary>
+    /// Finishes creating the parameter, and returns the original builder.
+    /// </summary>
+    /// <returns>The builder this parameter belongs to.</returns>
+    public CommandBuilder Finish()
+    {
+        return _builder;
+    }
+
+    /// <summary>
+    /// Builds the current builder into a <see cref="IParameterShape"/>.
+    /// </summary>
+    /// <returns>The constructed <see cref="IParameterShape"/>.</returns>
+    public IParameterShape Build()
+    {
+        var rangeAttribute = _attributes.OfType<RangeAttribute>().SingleOrDefault();
+        var optionAttribute = _attributes.OfType<OptionAttribute>().SingleOrDefault();
+
+        if (optionAttribute is SwitchAttribute)
+        {
+            if (_defaultValue is not bool || _parameterType != typeof(bool))
+            {
+                throw new InvalidOperationException
+                (
+                 $"A switch parameter must be a {typeof(bool)}" +
+                 $" (The parameter \"{_name}\" was marked as ({_parameterType})."
+                );
+            }
+        }
+
+        return optionAttribute is null
+                   ? CreatePositionalParameterShape(rangeAttribute)
+                   : CreateNamedParameterShape(optionAttribute, rangeAttribute);
+    }
+
+    private IParameterShape CreateNamedParameterShape
+    (
```

Comment:
Check if you really need to break these (the private ones) method declarations over multiple lines - 120 columns is the limit, not 80.

Reply:
They did in fact go over the line: 
![image](https://github.com/Remora/Remora.Commands/assets/42438262/9c6038cb-c031-4fcc-9821-b8855bd51747)